### PR TITLE
move all import/export functionality into new 'reports' plugin

### PR DESCRIPTION
**Pull request by binary1230**
see https://github.com/magfest/reports/pull/1 for discussion

merge at the same time as: 
https://github.com/magfest/reports/pull/1
https://github.com/magfest/production-config/pull/131
https://github.com/magfest/simple-rams-deploy/pull/22/files

### CODE REVIEW

**Comment by kitsuta** [comment]
You need to also remove valid_attendees and all_attendees from summary.py.
They're aliases that I added in because they were the old links to the
export. You don't need to add them to the plugin because if they're in a
different section then people's old bookmarks won't work anyway.

**Comment by EliAndrewC** [comment]
+1

**Comment by binary1230** [comment]
good call on the aliased functions.  I am looking into doing something like this here to create aliases: https://github.com/magfest/reports/pull/3

**Comment by kitsuta** [comment]
Sounds good. The aliases were a nicety for some folks who have had to deal with changing URLs as we improve and update the functionality of the exports, so it's not required that they be kept in the reports repo.

File: uber/decorators.py [hunk 1]
```diff
@@ -267,12 +285,12 @@ def screw_you_nick(rendered, template):
         return rendered
 
 
-def _get_module_name(class_or_func):
+def get_module_name(class_or_func):
     return class_or_func.__module__.split('.')[-1]
```
Comment (binary1230) [inline uber/decorators.py]: no changes here, just making this non-private

File: uber/decorators.py [hunk 2]
```diff
@@ -337,16 +355,24 @@ def with_restrictions(*args, **kwargs):
...
 
+def set_renderable(func, acccess):
+    """
+    Return a function that is flagged correctly and is ready to be called by cherrypy as a request
+    """
+    func.restricted = getattr(func, 'restricted', acccess)
+    new_func = timed(cached_page(sessionized(restricted(renderable(func)))))
+    new_func.exposed = True
+    return new_func
+
```
Comment (binary1230) [inline uber/decorators.py]: I needed to extract this from ```@all_renderable``` in order to call it from the new redirect decorators

File: uber/decorators.py [hunk 3]
```diff
@@ -337,16 +355,24 @@ def with_restrictions(*args, **kwargs):
...
         self.needs_access = needs_access
 
     def __call__(self, klass):
         for name, func in klass.__dict__.items():
             if hasattr(func, '__call__'):
-                func.restricted = getattr(func, 'restricted', self.needs_access)
-                new_func = timed(cached_page(sessionized(restricted(renderable(func)))))
-                new_func.exposed = True
+                new_func = set_renderable(func, self.needs_access)
                 setattr(klass, name, new_func)
```
Comment (binary1230) [inline uber/decorators.py]: refactored into ```set_renderable()```

File: uber/site_sections/summary.py [hunk 4]
```diff
@@ -183,141 +183,13 @@ def food_eligible(self, session):
...
-    def csv_export(self, message='', **params):
-        if 'model' in params:
-            self.export_model(selected_model=params['model'])
-
-        return {
-            'message': message,
-            'tables': sorted(model.__name__ for model in Session.all_models())
-        }
-    csv_export.restricted = [c.ACCOUNTS and c.STATS and c.PEOPLE and c.MONEY]
-
```
Comment (binary1230) [inline uber/site_sections/summary.py]: moved without changes into the reports plugin

File: uber/site_sections/summary.py [hunk 5]
```diff
@@ -183,141 +183,13 @@ def food_eligible(self, session):
...
-                    # Also you should fill in whatever actual format you want.
-                    val = getattr(attendee, col.name)
-                    row.append(val.strftime('%Y-%m-%d %H:%M:%S') if val else '')
-                else:
-                    # For everything else we'll just dump the value, although we might
-                    # consider adding more special cases for things like foreign keys.
-                    row.append(getattr(attendee, col.name))
-            out.writerow(row)
-    export_model.restricted = [c.ACCOUNTS and c.STATS and c.PEOPLE and c.MONEY]
-
```
Comment (binary1230) [inline uber/site_sections/summary.py]: move without changes to the reports plugin

File: uber/templates/summary/csv_export.html [hunk 6]
```diff
@@ -1,20 +0,0 @@
...
-
-<form action="export_model" method="post" enctype="multipart/form-data">
-    <select name="selected_model">
-        {% options tables 'attendee' %}
-    </select><br />
-    <input type="submit" value="Download" />
-</form>
-
-
-{% endblock %}
```
Comment (binary1230) [inline uber/templates/summary/csv_export.html]: moved to reports plugin

File: uber/templates/summary/csv_import.html [hunk 7]
```diff
@@ -1,44 +0,0 @@
...
-        </td>
-        <td id="name">{{ attendee|form_link }}</td>
-        <td id="type">{{ attendee.type }} {{ attendee.badge_type_label }}</td>
-        <td id="when">{{ attendee.when }} {{ attendee.registered }}</td>
-    </tr>
-{% endfor %}
-</table>
-{% endif %}
-
-{% endblock %}
```
Comment (binary1230) [inline uber/templates/summary/csv_import.html]: moved to reports plugin

**Pull request merged**